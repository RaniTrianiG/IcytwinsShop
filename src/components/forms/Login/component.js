import React from 'react';
import { Field } from 'redux-form';
import { View, Text } from 'react-native';
import { Button } from 'react-native-elements';
import PropTypes from 'prop-types';
import InputField from '../../elements/TextInput';
import CheckInput from '../../elements/CheckInput';
import style from './styles';

const Component = props => {
  const { handleSubmit } = props;
  return (
    <View>
      <Field component={InputField} name="username" placeholder="Email" label="Email" />
      <Field component={InputField} name="password" placeholder="Password" label="Password" secureTextEntry />
      <View style={style.checkboxContainer}>
        <Field component={CheckInput} name="remember" label="Ingat Saya" />
        <Text>Lupa Password?</Text>
      </View>
      <Button onPress={handleSubmit} title="Masuk" style={{ color: 'red' }} />
    </View>
  );
};

Component.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default Component;
