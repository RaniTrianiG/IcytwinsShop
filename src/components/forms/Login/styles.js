import { StyleSheet } from 'react-native';

const style = StyleSheet.create({
  checkboxContainer: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 15
  }
});

export default style;
