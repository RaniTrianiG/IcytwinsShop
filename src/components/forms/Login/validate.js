import validateInput from '../../../utils/validateInput';

const validate = values => {
  const { email, password } = values;

  const errors = {
    email: validateInput(email, [
      {
        rule: 'required',
        message: 'Email required'
      },
      {
        rule: 'email',
        message: 'Incorrect email format'
      }
    ]),
    password: validateInput(password, [
      {
        rule: 'required',
        message: 'Password required'
      }
    ])
  };
  return errors;
};

export default validate;
