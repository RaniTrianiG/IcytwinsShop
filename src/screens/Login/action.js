import jwtDecode from 'jwt-decode';
import AsyncStorage from '@react-native-community/async-storage';
import { ACTIONS, API } from '../../constants';
import { AXIOS } from '../../configs';
import { reqOption } from '../../utils';

export const postLogin = body => async dispatch => {
  dispatch({ type: ACTIONS.POST_LOGIN_START });
  const option = await reqOption();
  AXIOS.post(API.LOGIN, body, option)
    .then(resp => {
      const { data } = resp;
      const userData = jwtDecode(data.token);
      AsyncStorage.setItem('user-token', data.token);
      // startTabs();
      dispatch({ type: ACTIONS.POST_LOGIN_SUCCESS, data: userData });
    })
    .catch(error => {
      dispatch({ type: ACTIONS.POST_LOGIN_FAILED, error });
    });
};

export const autoLogin = token => dispatch => {
  const data = jwtDecode(token);
  dispatch({ type: ACTIONS.POST_LOGIN_SUCCESS, data });
};
