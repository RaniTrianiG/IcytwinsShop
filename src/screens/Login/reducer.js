import { ACTIONS } from '../../constants';
import { objectUpdater } from '../../utils';

const initialState = {
  isLoading: false,
  error: null,
  data: null,
  profile: null
};

const reducers = (state = initialState, action) => {
  const { data, error } = action;
  switch (action.type) {
    case ACTIONS.POST_LOGIN_START:
      return objectUpdater(state, { isLoading: true });
    case ACTIONS.POST_LOGIN_SUCCESS:
      return objectUpdater(state, { isLoading: false, data });
    case ACTIONS.POST_LOGIN_FAILED:
      return objectUpdater(state, { isLoading: false, error });
    case ACTIONS.SET_PROFILE_SUCCESS:
      return objectUpdater(state, { profile: data });
    default:
      return state;
  }
};

export default reducers;
