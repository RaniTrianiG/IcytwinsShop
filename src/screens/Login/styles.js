import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  anchor: {
    color: '#3088C8',
    textDecorationLine: 'underline'
  },
  footer: {
    alignSelf: 'center',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    paddingVertical: 15,
    width: 200
  },
  loginScreen: {
    flex: 1,
    padding: 10
  },
  logoContainer: {
    alignSelf: 'center',
    height: '100%',
    marginBottom: 35,
    width: '100%'
  }
});

export default styles;
