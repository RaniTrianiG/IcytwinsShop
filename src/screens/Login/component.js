import React from 'react';
import { View, Text } from 'react-native';
import { Navigation } from 'react-native-navigation';
import PropTypes from 'prop-types';
import FormLogin from '../../components/forms/Login';
import { SCREENS } from '../../constants';
import styles from './styles';
import Logo from '../../assets/svg/login-illustration.svg';

class Login extends React.Component {
  _handleSubmit = data => {
    const { actions } = this.props;
    actions.postLogin(data);
  };

  _handleRegister = () => {
    const { componentId } = this.props;
    Navigation.push(componentId, {
      component: {
        name: SCREENS.REGISTER,
        options: {
          statusBar: {
            style: 'dark',
            backgroundColor: '#f2f2f2'
          }
        }
      }
    });
  };

  render() {
    return (
      <View style={styles.loginScreen}>
        <View
          style={{
            height: 250,
            alignItems: 'center'
          }}
        >
          <Logo width="100%" height="100%" />
        </View>
        <FormLogin onSubmit={this._handleSubmit} />
        <View style={styles.footer}>
          <Text>Belum punya akun, </Text>
          <Text onPress={this._handleRegister} style={styles.anchor}>
            Daftar disini
          </Text>
        </View>
      </View>
    );
  }
}

Login.defaultProps = {
  componentId: 'loginscreen',
  actions: {}
};

Login.propTypes = {
  componentId: PropTypes.string,
  actions: PropTypes.object
};

export default Login;
