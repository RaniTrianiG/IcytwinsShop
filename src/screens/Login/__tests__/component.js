import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { reducer as formReducer } from 'redux-form';

import renderer from 'react-test-renderer';
import Login from '../component';

jest.mock('@react-native-community/netinfo');
jest.mock('react-native-navigation');

describe('Login Tests', () => {
  const mockStore = configureMockStore([thunk]);
  let initialState;
  let store;
  let props;
  let wrapper;
  beforeEach(() => {
    props = {
      actions: {
        postLogin: jest.fn(),
        autoLogin: jest.fn()
      },
      isLoading: false
    };
    initialState = {
      form: formReducer
    };
    wrapper = shallow(<Login {...props} />);
  });

  afterEach(() => {
    store.clearActions();
  });

  it('renders correctly with initial state', () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Login {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test go to register screen action', () => {
    const spy = jest.spyOn(wrapper.instance(), '_handleRegister');
    wrapper.instance().forceUpdate();
    wrapper
      .find('Text')
      .at(1)
      .simulate('press');
    expect(spy).toBeCalled();
  });

  it('test simulate submit action', () => {
    const spy = jest.spyOn(wrapper.instance(), '_handleSubmit');
    wrapper.instance().forceUpdate();
    wrapper.find('ReduxForm').simulate('submit');
    expect(spy).toBeCalled();
  });
});
