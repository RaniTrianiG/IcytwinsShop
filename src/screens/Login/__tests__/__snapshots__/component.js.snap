// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Tests renders correctly with initial state 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 250,
      }
    }
  >
    <SvgMock
      height="100%"
      width="100%"
    />
  </View>
  <View>
    <View
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 10,
            "marginLeft": 10,
          }
        }
      >
        Email
      </Text>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#ccc",
              "borderRadius": 5,
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 5,
              "padding": 3,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            name="username"
            onBlur={[Function]}
            onChangeText={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Email"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 10,
            "marginLeft": 10,
          }
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#ccc",
              "borderRadius": 5,
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 5,
              "padding": 3,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            name="password"
            onBlur={[Function]}
            onChangeText={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 15,
        }
      }
    >
      <View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fafafa",
              "borderColor": "transparent",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "marginLeft": 0,
              "marginRight": 10,
              "opacity": 1,
              "padding": 0,
            }
          }
          testID="checkbox"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  Object {
                    "minWidth": 24,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï‚–
            </Text>
            <Text
              style={
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="checkboxTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Ingat Saya
            </Text>
          </View>
        </View>
      </View>
      <Text>
        Lupa Password?
      </Text>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "color": "red",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Masuk
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 15,
        "width": 200,
      }
    }
  >
    <Text>
      Belum punya akun, 
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#3088C8",
          "textDecorationLine": "underline",
        }
      }
    >
      Daftar disini
    </Text>
  </View>
</View>
`;
